### PR TITLE
[#2874] Instructor help page revamp

### LINKED ISSUE #2874: Instructor help page: mention the 'moderation' feature
also mention sending reminders individually

### PR DESCRIPTION
**Pull request by fans2619**
Part of #2874 
Part of #8769 
Part of #11794

### CODE REVIEW

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 1]
```diff
@@ -185,6 +185,37 @@ <h4 *ngIf="displaySubsection(0, 7)">Setting Up Sessions</h4>
...
+          <div class="row padding-15px text-left">
+            <div class="col-md-2">
+              <div class="question-basic-info">[Optional]<br/>Description</div>
+            </div>
+            <div class="col-md-10">
+              <tm-rich-text-editor [richText]="'<p><strong>This question is </strong><span style=\'color: rgb(255, 0, 0); background-color: rgb(255, 255, 0);\'><strong>compulsory</strong></span><strong>!</strong></p>'" minHeightInPx="100"></tm-rich-text-editor>
+            </div>
+          </div>
+        </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:215]: Highly discouraged; leads to things like #11392. But if you're still too unfamiliar with Angular to be able to go one step further and extract out the necessary component(s), I can close one eye.

[the remark this answers is not in the document]
Comment (fans2619) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:215]: Ah okay if it's highly discouraged, then I can try extracting.

[the remark this answers is not in the document]
Comment (fans2619) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:215]: @damithc Do you think this UI is fine?

[the remark this answers is not in the document]
Comment (fans2619) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:215]: Almost done actually. I'll make a PR later.

**Comment by fans2619** [comment]
I'll add some screenshots later.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 2]
```diff
@@ -185,6 +185,35 @@ <h4 *ngIf="displaySubsection(0, 7)">Setting Up Sessions</h4>
...
+          <div class="col-md-2">
+            <div class="question-basic-info">[Optional]<br/>Description</div>
+          </div>
+          <div class="col-md-10">
+            <tm-rich-text-editor [richText]="'<p><strong>This question is </strong><span style=\'color: rgb(255, 0, 0); background-color: rgb(255, 255, 0);\'><strong>compulsory</strong></span><strong>!</strong></p>'" minHeightInPx="100"></tm-rich-text-editor>
+          </div>
+        </div>
+      </div>
+    </tm-example-box>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:215]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 3]
```diff
@@ -217,46 +246,68 @@ <h4 *ngIf="displaySubsection(0, 7)">Setting Up Sessions</h4>
...
-        They should specify the exact email address that was used to enroll them in TEAMMATES.
+        Use the <button class="btn btn-sm btn-secondary">Remind</button> button on your <b>Home</b> or <b>Sessions</b> page (there is one for each session) to resend emails to specific students or all those who haven't submitted yet. Also, ask students to check their spam box.
       </li>
       <li>
-        Failing all above, ask students to email us at <a href="mailto:{{ supportEmail }}">{{ supportEmail }}</a> We can resend the email using an alternate means.
+        Failing all above, ask students to email us at <a href="mailto:{{ supportEmail }}">{{ supportEmail }}</a>. We can resend the email using an alternate means.
         When they email us, they should use the email address currently registered in TEAMMATES (for identification).
       </li>
     </ol>
   </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:274]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 4]
```diff
@@ -217,46 +246,68 @@ <h4 *ngIf="displaySubsection(0, 7)">Setting Up Sessions</h4>
...
+                            headerText="Can I change the visibility settings of questions after the session has started?"
+                            [(isPanelExpanded)]="questionsToCollapsed[SessionsSectionQuestions.CHANGE_VISIBILITY_AFTER_SESSION_START]">
+    <p>
+      Yes, but it is not recommended.
+    </p>
+    <p>
+      Reason: The current visibility setting of each question is shown to the respondent at the point of response submission i.e., responses are submitted based on the promise of a certain visibility.
+      Changing the visibility later would be like 'breaking a promise' you made earlier.
+    </p>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:296]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 5]
```diff
@@ -217,46 +246,68 @@ <h4 *ngIf="displaySubsection(0, 7)">Setting Up Sessions</h4>
...
+  <h4 *ngIf="displaySubsection(9, 17)">Managing Session Responses</h4>
+  <!-- Question -->
+  <tm-instructor-help-panel class="instr-help-qn-first" #question [hidden]="key && !showQuestion.includes(SessionsSectionQuestions.STUDENT_EDIT_RESPONSE)"
+                            [id]="SessionsSectionQuestions.STUDENT_EDIT_RESPONSE"
+                            headerText="Can students edit responses they submitted earlier?"
+                            [(isPanelExpanded)]="questionsToCollapsed[SessionsSectionQuestions.STUDENT_EDIT_RESPONSE]">
+    <p>
+      Yes, they can edit responses any number of times, as long as the session is still open. To edit responses, they can use the same method they used when submitting the responses initially.
+    </p>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:308]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 6]
```diff
@@ -281,7 +332,16 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
...
-  <!--Question-->
+  <!-- Question -->
+  <tm-instructor-help-panel #question [hidden]="key && !showQuestion.includes(SessionsSectionQuestions.STUDENT_ACCESS_SUBMISSION_PAGE)"
+                            [id]="SessionsSectionQuestions.STUDENT_ACCESS_SUBMISSION_PAGE"
+                            headerText="Can I see when a student tried to access the submission page?"
+                            [(isPanelExpanded)]="questionsToCollapsed[SessionsSectionQuestions.STUDENT_ACCESS_SUBMISSION_PAGE]">
+    <p>
+      Not at the moment. If there is such a need, email us at <a href="mailto:{{ supportEmail }}">{{ supportEmail }}</a>, and we'll try to find that information for you.
+    </p>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:343]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 7]
```diff
@@ -297,8 +357,7 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
       <tm-view-results-panel></tm-view-results-panel>
     </tm-example-box>
     <p>
-      In the example below, results are sorted by <b>Giver > Recipient > Question</b>. Additionally, missing responses
-      have been recorded.
+      In the example below, results are sorted by <b>Giver &#8594; Recipient &#8594; Question</b>. Additionally, missing responses have been recorded.
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:360]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 8]
```diff
@@ -343,26 +402,82 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
...
+        <a (click)="jumpTo(SessionsSectionQuestions.SESSION_VIEW_RESULTS); questionsToCollapsed[SessionsSectionQuestions.SESSION_VIEW_RESULTS] = true" [tmRouterLink]="">View the results</a> of the session.
+      </li>
+      <li>
+        Locate the response you would like to moderate and click the <button class="btn btn-light btn-sm">Moderate response</button> button associated with the response.
+      </li>
+      <li>
+        Edit the response as if you were the submitter. Remember to click the <button class="btn btn-success btn-sm">Submit Responses</button> button after your edit.
+      </li>
+    </ol>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:427]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 9]
```diff
@@ -343,26 +402,82 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
...
+  <!-- Question -->
+  <tm-instructor-help-panel #question [hidden]="key && !showQuestion.includes(SessionsSectionQuestions.STUDENT_SEE_RESPONSE)"
+                            [id]="SessionsSectionQuestions.STUDENT_SEE_RESPONSE"
+                            headerText="When do students get to see the results collected by a session?"
+                            [(isPanelExpanded)]="questionsToCollapsed[SessionsSectionQuestions.STUDENT_SEE_RESPONSE]">
+    <p>
+      Students can only see the results after you have published the results of that session. Until then, they can only see responses submitted by themselves.<br>
+      When you publish the results of a session, TEAMMATES will email each student instructions for viewing results. What students will get to see depends on the visibility setting of each question.
+    </p>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:437]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 10]
```diff
@@ -343,26 +402,82 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
...
+      <li>
+        Go to the <b>Home</b> or <b>Sessions</b> page and click the <button class="btn btn-light btn-sm dropdown-toggle">Results</button> button of the published session.<br>
+        Then select <b>Resend link to view results</b> in the drop-down menu to resend emails to specific students. Make sure the session results are published if you are unable to see the option.
+      </li>
+      <li>
+        Failing all above, ask students to email us at <a href="mailto:{{ supportEmail }}">{{ supportEmail }}</a>. We can resend the email using an alternate means.
+        When they email us, they should use the email address currently registered in TEAMMATES (for identification).
+      </li>
+    </ol>
+  </tm-instructor-help-panel>
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:460]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html [hunk 11]
```diff
@@ -343,26 +402,82 @@ <h4 *ngIf="displaySubsection(7, 10)">Managing Session Responses</h4>
...
       To create comments on a response in a session:
     </p>
     <ol>
       <li>
-        View the results of a session.
+        <a (click)="jumpTo(SessionsSectionQuestions.SESSION_VIEW_RESULTS); questionsToCollapsed[SessionsSectionQuestions.SESSION_VIEW_RESULTS] = true" [tmRouterLink]="">View the results</a> of the session.
       </li>
       <li>
-        Click <button class="btn btn-light btn-sm dropdown-toggle">View</button> dropdown and change the view type to <b>Group by - Giver &gt; Recipient &gt; Question</b> or <b>Group by - Recipient &gt; Giver &gt; Question</b>.
+        Click the <button class="btn btn-light btn-sm dropdown-toggle">View</button> dropdown and change the view type to <b>Group by - Giver &#8594; Recipient &#8594; Question</b> or <b>Group by - Recipient &#8594; Giver &#8594; Question</b>.
```
Comment (fans2619) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/instructor-help-sessions-section.component.html:480]: image

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/sessions-section-questions.ts [hunk 12]
```diff
@@ -73,13 +73,49 @@ export enum SessionsSectionQuestions {
...
+  STUDENT_DID_NOT_RECEIVE_SUBMISSION_LINK = 'student-did-not-receive-submission-link',
 
   /**
    * How do I extend the deadline of a session?
    */
   EXTEND_SESSION_DEADLINE = 'extend-session-deadline',
 
+  /**
+   * Can I set a question as ‘compulsory’?
+   */
```
Comment (jianhandev) [inline src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/sessions-section-questions.ts:87]:
```suggestion
   * Can I set a question as 'compulsory'?
   */
```

**Review by jianhandev — COMMENTED** [review]
Changes look quite ok to me, and thanks for all the screenshots. As this PR addresses quite a few instructor help tips more than just the 'moderation feature', the scope of the PR title could be broader? Also I would suggest adding to the PR description:
* Which new questions are added
* Changes to outdated help tips (if not trivial)

e.g. 
New questions added to instructor help page:
- [x] Can I set a question as 'compulsory'?
- [x] Can students edit responses they submitted earlier?
- [ ] etc...

Could also use something like this to keep track of small todos in your future PRs.

File: src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/sessions-section-questions.ts (hunk 12 above)
Comment (fans2619) [reply src/web/app/pages-help/instructor-help-page/instructor-help-sessions-section/sessions-section-questions.ts:87]: Thanks!

**Comment by fans2619** [comment]
> As this PR addresses quite a few instructor help tips more than just the 'moderation feature', the scope of the PR title could be broader?

Hi actually I'm not sure if I'm allowed to use a title different from one of the issues'.

> Also I would suggest adding to the PR description:

Actually I used the Google docs as my reference when working on this PR. The link is in the Slack #teammates channel and this PR addresses the questions at the bottom of the document. I can copy the questions here some time later as well!

> Could also use something like this to keep track of small todos in your future PRs.

Okay sure!

**Comment by wkurniawan07** [comment]
> I'm not sure if I'm allowed to use a title different from one of the issues'.

<https://teammates.github.io/teammates/process.html#step-4-submit-a-pr>
If you're only solving an issue partially, you're in fact encouraged to describe what is done instead of copy-pasting the issue.

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**